google/cloud/forseti/services/inventory/base/resources.py

```diff
@@ -2194,6 +2271,163 @@ class KubernetesClusterIterator(resource_iter_class_factory(
+
+    def iter(self):
+        """Resource iterator.
+
+        Yields:
+            Resource: KubernetesCluster created
+        """
+        gcp = self.client
+        try:
+            for data, metadata in gcp.iter_k8s_pods(
+                    project_id=self.resource.parent().parent()['id'],
+                    zone=self.resource['zone'],
+                    cluster=self.resource['name'],
+                    namespace=self.resource['namespace']):
+                yield FACTORIES['kubernetes_pod'].create_new(
+                    data, metadata=metadata)
+        except ResourceNotSupported as e:
+            # API client doesn't support this resource, ignore.
+            LOGGER.debug(e)
+
+
+class KubernetesNamespaceIterator(ResourceIterator):
+    """The Resource iterator implementation for KubernetesCluster"""
+
+    def iter(self):
+        """Resource iterator.
+
+        Yields:
+            Resource: KubernetesCluster created
+        """
+        gcp = self.client
+        try:
+            for data, metadata in gcp.iter_k8s_namespaces(
+                    project_id=self.resource.parent().parent()['projectId'],
+                    zone=self.resource['zone'],
+                    cluster=self.resource['name']):
+                yield FACTORIES['kubernetes_namespace'].create_new(
+                    data, metadata=metadata)
+        except ResourceNotSupported as e:
+            # API client doesn't support this resource, ignore.
+            LOGGER.debug(e)
+
+
+class KubernetesRoleIterator(ResourceIterator):
+    """The Resource iterator implementation for KubernetesCluster"""
+
+    def iter(self):
+        """Resource iterator.
+
+        Yields:
+            Resource: KubernetesCluster created
+        """
+        gcp = self.client
+        try:
+            for data, metadata in gcp.iter_k8s_roles(
+                    project_id=self.resource.parent().parent()['projectId'],
+                    zone=self.resource['zone'],
+                    cluster=self.resource['name'],
+                    namespace=self.resource['namespace']):
+                yield FACTORIES['kubernetes_role'].create_new(
+                    data, metadata=metadata)
+        except ResourceNotSupported as e:
+            # API client doesn't support this resource, ignore.
+            LOGGER.debug(e)
+
+
+class KubernetesRoleBindingIterator(ResourceIterator):
+    """The Resource iterator implementation for KubernetesCluster"""
+
+    def iter(self):
+        """Resource iterator.
+
+        Yields:
+            Resource: KubernetesCluster created
+        """
+        gcp = self.client
+        try:
+            for data, metadata in gcp.iter_k8s_rolebindings(
+                    project_id=self.resource.parent().parent()['projectId'],
+                    zone=self.resource['zone'],
+                    cluster=self.resource['name'],
+                    namespace=self.resource['namespace']):
```

Comment:
The namespace name is in the ['metadata']['name'] key of the Namespace resource.

So it should be namespace=self.resource['metadata']['name']

Same below.

Reply:
I see. Thanks for the explanation! Updated.